assembly/src/parsers/field_ops.rs

```diff
@@ -178,5 +178,5 @@ fn parse_bit_len_param(op: &Token, param_idx: usize) -> Result<u8, ParsingError>
 }
 
 fn parse_imm_value(op: &Token) -> Result<Felt, ParsingError> {
-    Ok(Felt::new(parse_checked_param::<u64>(op, 1, 0, Felt::MODULUS - 1)?))
+    Ok(Felt::new(parse_checked_param::<u64, _>(op, 1, 0..=(Felt::MODULUS - 1))?))
```

Comment:
optional nit: instead of making an inclusive range check - 1, we can `0..Felt::MODULUS`